geoapps_utils/driver/data.py

```diff
@@ -56,8 +56,10 @@ def collect_input_from_dict(
         """
         update = {}
         for field, info in base_model.model_fields.items():
-            if isinstance(info.annotation, type) and issubclass(
-                info.annotation, BaseModel
+            if (
+                isinstance(info.annotation, type)
+                and not isinstance(info.annotation, GenericAlias)
```

Comment:
What's the reason for no Generic?

Reply:
It's because of pydantic Annotations if I remember well.
There was a bug at this place because of that

Reply:
Hmm ok, not clear what's up. I ll move it to test anyway, we have time to revise later.